File: client/src/types/Passenger.ts
export interface Passenger {
  PassengerId: number;
  Name: string | null;
  Survived: number | null;
  Pclass: number | null;
  Sex: string | null;
  Age: number | null;
  SibSp?: number;
  Parch?: number;
  Ticket?: string;
  Fare?: number | null;
  Cabin?: string | null;
  Embarked?: string | null;
}

export type PassengerColumn = 
  'Survived' | 
  'Pclass' | 
  'Sex' | 
  'Age' | 
  'SibSp' | 
  'Parch' | 
  'Ticket' | 
  'Fare' | 
  'Cabin' | 
  'Embarked';

File: api/app/models/passenger.py
from sqlmodel import SQLModel, Field

class Passenger(SQLModel, table=True):
    PassengerId: int = Field(primary_key=True)
    Name: str | None = None
    Survived: int | None = None
    Pclass: int | None = None
    Sex: str | None = None
    Age: float | None = None
    SibSp: int | None = None
    Parch: int | None = None
    Ticket: str | None = None
    Fare: float | None = None
    Cabin: str | None = None
    Embarked: str | None = None